mrpt_rbpf_slam/include/mrpt_rbpf_slam/mrpt_rbpf_slam.h

```diff
@@ -83,9 +83,9 @@ class PFslam
                    const mrpt::obs::CObservationOdometry::ConstPtr odometry);
 
 protected:
-  mrpt::slam::CMetricMapBuilderRBPF* mapBuilder_;  ///< map builder
-  mrpt::obs::CActionCollection::Ptr action_;       ///< actions
-  mrpt::obs::CSensoryFrame::Ptr sensory_frame_;    ///< observations
+  mrpt::slam::CMetricMapBuilderRBPF mapBuilder_;  ///< map builder
```

Comment:
Better, no doubt...

Reply:
Yes, I meant that, whenever possible, one should prefer stack variables, of course! ;-)
::Ptr are only defined for classes inheriting from mrpt::rtti::CObject... or a few others which are manually added for consistency. 
If you see it useful, don't doubt adding a PR to mrpt/mrpt to add missing `using Ptr = std::shared_ptr<T>;` where needed.

Reply:
@jlblancoc, somehow this answer flew under my radar, so I didn't see it until now. So the basic idea in mrpt is to have Ptr alias only for polymorphic objects? That makes sense. 

A bit of a digression, but maybe you've seen this before, I was actually wondering if anyone is defining a similar alias for a unique_ptr?

Reply:
No problem! :-)

Actually, the Ptr alias is there for any class inheriting from CObject... so, all are strictly speaking polymorphic wrt CObject.

And yes: in mrpt2, you have ::Ptr, ::ConstPtr, ::UniquePtr, etc. Also: ::Create(), ::CreateUnique(),... 

Any help improving the related docs (which are quite poor!), would be welcome :-)
https://docs.mrpt.org/reference/master/group__mrpt__rtti__grp.html